File: app/src/main/java/com/example/meeera/imageconverter/FileSaveRepository.java
package com.example.meeera.imageconverter;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface FileSaveRepository {
    @Insert
    long save(FileSaveModel fileSaveModel);

    @Query("SELECT * FROM FileSaveModel")
    List<FileSaveModel> getSavedFiles();
}

File: app/src/main/java/com/example/meeera/imageconverter/FileSaveRepository.kt
package com.example.meeera.imageconverter


import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Transaction

@Dao
interface FileSaveRepository {

    @Query("SELECT * FROM FileSaveModel")
    suspend fun savedFiles(): List<FileSaveModel>

    @Insert
    suspend fun save(fileSaveModel: FileSaveModel): Long
}
